### ISSUE TITLE
Doesn't validate JSDoc comments that aren't above a function definition

### ISSUE THREAD

**Issue by snoack** [gajus/eslint-plugin-jsdoc#253]
The `valid-types` rule doesn't identify the missing `>` in the type expression below, yet when running jsdoc on that source code it fails.

```javascript
class Matcher
{
  constructor()
  {
    ...

    /**
     * Lookup table of compiled patterns for simple filters by their associated
     * keyword
     * @type {Map.<string,?CompiledPatterns}
     * @private
     */
    this._compiledPatternsByKeyword = new Map();
  }
}
```

**Comment by brettz9** [comment]
The issue in this case is actually that `valid-types` only checks comment blocks above function scopes, so it isn't checking that block at all (but when placing such a type on the function instead, it handles the type ok). I'll close this therefore, but you can subscribe to #214 which is tracking the desire to see the scope expanded, e.g., to adding above properties/variables as in your case.
